cli/internal/core/engine.go

```diff
@@ -383,3 +384,41 @@ func (e *Engine) ValidatePersistentDependencies(graph *graph.CompleteGraph) erro
 func (e *Engine) GetResolvedTaskDefinition(rootPipeline *fs.Pipeline, taskName string, taskID string) (*fs.TaskDefinition, error) {
 	return rootPipeline.GetTask(taskID, taskName)
 }
+
+// GetTaskGraphAncestors gets all the ancestors for a given task in the graph.
+// "Ancestors" are all tasks that the given task depends on.
+// This is only used by DryRun output right now.
+func (e *Engine) GetTaskGraphAncestors(taskID string) ([]string, error) {
+	ancestors, err := e.TaskGraph.Ancestors(taskID)
+	if err != nil {
+		return nil, err
+	}
+	stringAncestors := []string{}
+	for _, dep := range ancestors {
+		// Don't leak out internal ROOT_NODE_NAME nodes, which are just placeholders
+		if !strings.Contains(dep.(string), ROOT_NODE_NAME) {
+			stringAncestors = append(stringAncestors, dep.(string))
+		}
+	}
+	// TODO(mehulkar): Why are ancestors not sorted, but GetTaskGraphDescendants sorts?
+	return stringAncestors, nil
+}
+
+// GetTaskGraphDescendants gets all the ancestors for a given task in the graph.
```

Comment:
Should this be 'gets all the descendants'?

Reply:
it absolutely should. good catch 🤦🏾